src/components/Cart.js

```diff
@@ -0,0 +1,112 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useEffect, useState } from "react";
+
+export default function Cart({cart, setCart}) {
+
+    const [purchaseValue, setPurchaseValue] = useState(0);
+
+    function handleUpdateQuantity(value, key){
+        setCart((prev)=>({...prev,[key]:{...prev[key], quantity:value}}));
+    }
+
+    function handleDeleteCartItem(key){
+        setCart((prev)=>{ delete prev[key]; return {...prev};});
+    }
+    
+    useEffect(()=>{
+        var cartValue = 0;
```

Comment:
@rahulmeghwal Is there a cleaner way to sum up the values here, maybe taking an immutable, functional approach so you don't need to redefine a variable (my understanding at least is the functional approach is preferable in the front-end/React world)?

Reply:
Have updated the code to use a functional approach to sum up cart value using map and reduce.